test/python/v1/nexus/test_nexus.py

```diff
@@ -0,0 +1,518 @@
+        hdls["ms2"].replica_destroy(UUID)
+    except Exception as e:
+        logging.debug(e)
+
+
+def test_enospace_on_volume(mayastors, create_replica):
+    nodes = mayastors
+    pools = []
+    uuid = guid.uuid4()
+
+    pools.append(nodes["ms2"].pools_as_uris()[0])
+    pools.append(nodes["ms1"].pools_as_uris()[0])
+    nexus_node = nodes["ms3"].as_target()
+
+    v = Volume(uuid, nexus_node, pools, 100 * 1024 * 1024)
+
+    with pytest.raises(grpc.RpcError) as error:
+        v.create()
+    assert error.value.code() == grpc.StatusCode.RESOURCE_EXHAUSTED
+
+
+async def kill_after(container, sec):
+    """Kill the given container after sec seconds."""
+    await asyncio.sleep(sec)
+    logging.info(f"killing container {container}")
+    container.kill()
+
+
+@pytest.mark.asyncio
+async def test_nexus_2_mirror_kill_one(containers, mayastors, create_nexus):
+
+    uri = create_nexus
+    nvme_discover(uri)
+    dev = nvme_connect(uri)
+    try:
+        job = Fio("job1", "rw", dev).build()
+        print(job)
+
+        to_kill = containers.get("ms2")
+        await asyncio.gather(run_cmd_async(job), kill_after(to_kill, 5))
+
+    finally:
+        # disconnect target before we shutdown
+        nvme_disconnect(uri)
+
+
+@pytest.mark.asyncio
+async def test_nexus_2_remote_mirror_kill_one(
+    containers, mayastors, nexus_uuid, create_nexus
+):
+    """
+    This test does the following steps:
+
+        - creates mayastor instances
+        - creates pools on mayastor 1 and 2
+        - creates replicas on those pools
+        - creates a nexus on mayastor 3
+        - starts fio on a remote VM (vixos1) for 15 secondsj
+        - kills mayastor 2 after 4 seconds
+        - assume the test to succeed
+        - disconnect the VM from mayastor 3 when FIO completes
+        - removes the nexus from mayastor 3
+        - removes the replicas but as mayastor 2 is down, will swallow errors
+        - removes the pool
+
+    The bulk of this is done by reusing fixtures those fitures are not as
+    generic as one might like at this point so look/determine if you need them
+    to begin with.
+
+    By yielding from fixtures, after the tests the function is resumed where
+    yield is called.
+    """
+
+    uri = create_nexus
+    dev = nvme_connect(uri)
+    try:
+        job = Fio("job1", "randwrite", dev).build()
+        print(job)
+
+        to_kill = containers.get("ms2")
+
+        # create an event loop polling the async processes for completion
+        await asyncio.gather(run_cmd_async(job), kill_after(to_kill, 4))
+
+        list = mayastors.get("ms3").nexus_list(None)
+
+        NEXUS_UUID, size_mb = nexus_uuid
+        nexus = next(n for n in list if n.uuid == NEXUS_UUID)
+
+        assert nexus.state == nexus_pb.NEXUS_DEGRADED
+        assert nexus.children[1].state == nexus_pb.CHILD_FAULTED
+
+    finally:
+        # disconnect target before we shutdown
+        nvme_disconnect(uri)
+
+
+# @pytest.mark.asyncio
```

Comment:
Is there a reason this test is commented out?

Reply:
No reason. I was unable to test these locally, forgot to uncomment these.

Reply:
Looks like it passed so that's fine.